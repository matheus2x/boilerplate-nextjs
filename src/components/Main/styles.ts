'use client';

import styled from 'styled-components';

export const Wrapper = styled.main`
  background-color: #17182b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;
