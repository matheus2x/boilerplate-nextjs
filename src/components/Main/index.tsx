import * as S from './styles';

const Main = ({
  title = 'Next.js Boilerplate',
  description = 'TS, React, Next.js, Styled Components, Storybook, Jest, Eslint, Prettier & Husky',
}) => (
  <S.Wrapper>
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
  </S.Wrapper>
);

export default Main;
