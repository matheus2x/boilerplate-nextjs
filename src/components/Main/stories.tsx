import { Meta, StoryObj } from '@storybook/react';

import Main from '.';

const meta: Meta<typeof Main> = {
  title: 'components/Main',
  component: Main,
  parameters: {
    latouy: 'fullscreen',
  },
};

export default meta;
export const Default: StoryObj = {};

export const Alternativo: StoryObj = {
  args: {
    title: 'Techs Frontend top',
    description: 'React, TS, Next.js, Storybook, etc..',
  },
};
