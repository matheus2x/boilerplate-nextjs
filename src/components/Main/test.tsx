import { render, screen } from '@testing-library/react';

import Main from '.';

describe('<Main />', () => {
  it('should render the reading', () => {
    // renderiza
    const { container } = render(<Main />);

    // busca elemento e verifica existencia
    expect(
      screen.getByRole('heading', { name: /Next.js React Boilerplate/i })
    ).toBeInTheDocument();

    // gerar snapshot
    expect(container.firstChild).toMatchSnapshot();
  });
});
