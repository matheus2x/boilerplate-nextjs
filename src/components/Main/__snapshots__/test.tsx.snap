// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the reading 1`] = `
.c0 {
  background-color: #17182b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.c1 {
  font-size: 2.5rem;
}

.c2 {
  font-size: 2rem;
  font-weight: 400;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Next.js Boilerplate
  </h1>
  <h2
    class="c2"
  >
    TS, React, Next.js, Styled Components, Storybook, Jest, Eslint, Prettier & Husky
  </h2>
</main>
`;
